KomaMRIBase/src/datatypes/phantom/motion/ArbitraryMotion.jl

```diff
@@ -2,13 +2,30 @@
 #       Interpolator{T,Degree,ETPType}, 
 #           Degree = Linear,Cubic.... 
 #           ETPType = Periodic, Flat...
-const LinearInterpolator = Interpolations.Extrapolation{
-    T,
-    1,
-    Interpolations.GriddedInterpolation{T,1,V,Gridded{Linear{Throw{OnGrid}}},Tuple{V}},
-    Gridded{Linear{Throw{OnGrid}}},
-    Periodic{Nothing},
-} where {T<:Real,V<:AbstractVector{T}}
+
+const Interpolator = Interpolations.GriddedInterpolation{
+    T,N,V,Itp,K
+} where {
+    T<:Real,
+    N,
+    V<:AbstractArray{T},
+    Itp<:Tuple{ Vararg{ Union{Interpolations.Gridded{Linear{Throw{OnGrid}}},Interpolations.NoInterp} } },
+    K<:Tuple{Vararg{AbstractVector{T}}},
+}
+
+function GriddedInterpolation(
+    nodes::NType,
+    A::AType
+) where {T<:Real, AType<:AbstractArray{T}, NType<:Tuple{Vararg{AbstractVector{T}}}}
+    Ns, _ = size(A)
+    if Ns > 1
```

Comment:
Can't we get rid of this check?

Reply:
Solved in [872d16a](https://github.com/JuliaHealth/KomaMRI.jl/pull/408/commits/872d16a09b844b0bc0a62f02cf0bdc61ffd5f2b3)